python/pyspark/sql/context.py

```diff
@@ -46,6 +46,7 @@
 
 if TYPE_CHECKING:
     from py4j.java_gateway import JavaObject
+    import pyarrow as pa
```

Comment:
I think we can not assume `pyarrow` is installed by default in Spark Classic

https://spark.apache.org/docs/latest/api/python/getting_started/install.html#dependencies

@HyukjinKwon

Reply:
I think it is OK inside when `TYPE_CHECKING` is True. There is another file in the repo that has had `import pyarrow as pa` inside an `if TYPE_CHECKING:` conditional since 2021: https://github.com/apache/spark/pull/34101/files#diff-a4f1631a18d1b4921b8727e1d78059a1014c433239eb107962c473c6466214e5R29

Reply:
yeah this is fine with TYPE_CHECKING